### DISCUSSION TITLE
TOTP request for user actions

### DISCUSSION THREAD

**Discussion by ITEXNIKJS** [supabase/supabase#21006] [Questions] [answered]
Is there an option to request and verify a user's TOTP code for certain actions?

Is it possible to implement this functionality at the level of supabase policies?

**Comment by mwoss** [comment, marked as answer]
By verification of certain actions do you mean for example verification of deleting an account or creating a new resource? If so, as far as I know, it's not possible by using Supabase alone. You would need a middle layer (your backend) where you handle such cases on your own. OTPs can be used only for user registration and signing in right now.
